src/modules/iotjs_module_crypto.c

```diff
@@ -371,24 +401,95 @@ JS_FUNCTION(Sha1Encode) {
     return jerry_create_undefined();
   }
 
-  unsigned char *sha1_ret = NULL;
-  size_t sha1_sz =
-      iotjs_sha1_encode(&sha1_ret,
-                        (const unsigned char *)iotjs_string_data(&user_str),
-                        iotjs_string_size(&user_str));
+  size_t sha_sz = 0;
+
+  unsigned char *sha_ret = NULL;
+
+  switch (type) {
+    case IOTJS_CRYPTO_SHA1: {
+      sha_sz =
+          iotjs_sha1_encode(&sha_ret,
+                            (const unsigned char *)iotjs_string_data(&user_str),
+                            iotjs_string_size(&user_str));
+      break;
+    }
+    case IOTJS_CRYPTO_SHA256: {
+#if !ENABLE_MODULE_TLS
+      return JS_CREATE_ERROR(COMMON, no_tls_err_str);
+#else  /* ENABLE_MODULE_TLS */
+      sha_sz = iotjs_sha256_encode(&sha_ret,
+                                   (const unsigned char *)iotjs_string_data(
+                                       &user_str),
+                                   iotjs_string_size(&user_str));
+      break;
+#endif /* !ENABLE_MODULE_TLS */
+    }
+  }
+
   iotjs_string_destroy(&user_str);
 
   jerry_value_t ret_val;
-  ret_val = iotjs_bufferwrap_create_buffer(sha1_sz);
+  ret_val = iotjs_bufferwrap_create_buffer(sha_sz);
   iotjs_bufferwrap_t *ret_wrap = iotjs_bufferwrap_from_jbuffer(ret_val);
-  memcpy(ret_wrap->buffer, sha1_ret, sha1_sz);
-  ret_wrap->length = sha1_sz;
+  memcpy(ret_wrap->buffer, sha_ret, sha_sz);
+  ret_wrap->length = sha_sz;
 
-  IOTJS_RELEASE(sha1_ret);
+  IOTJS_RELEASE(sha_ret);
   return ret_val;
 }
 
 
+JS_FUNCTION(RsaVerify) {
+#if !ENABLE_MODULE_TLS
+  return JS_CREATE_ERROR(COMMON, no_tls_err_str);
+#endif
+  DJS_CHECK_THIS();
+  DJS_CHECK_ARGS(2, any, any);
+
+  uint8_t type = JS_GET_ARG(0, number);
+  jerry_value_t jdata = JS_GET_ARG(1, any);
+  jerry_value_t jkey = JS_GET_ARG(2, any);
+  jerry_value_t jsignature = JS_GET_ARG(3, any);
+
+  iotjs_string_t key = iotjs_string_create();
```

Comment:
Tricky way of using `iotjs_jbuffer_as_string`. The function could be extended to reset the string when returns false.